lib/osf-components/addon/components/file-browser/template.hbs

```diff
@@ -2,94 +2,124 @@
-            @horizontalPosition='left'
-            as |dropdown|
-        >
-            <label>
-                {{t 'registries.overview.files.sort'}}
-                <dropdown.trigger
-                    data-test-file-sort-trigger
-                    data-analytics-name='Sort files'
-                    local-class='SortDropdown__button {{if this.isMobile 'Mobile'}}'
-                >
-                    <span local-class='SortedBy {{if this.isMobile 'Mobile'}}'>
-                        {{t (concat 'registries.overview.files.sort_by.' @manager.sort)}}
-                    </span>
-                    <FaIcon @icon='caret-down' />
-                </dropdown.trigger>
-            </label>
-            <dropdown.content
-                local-class='DropdownList'
-                {{on 'click' dropdown.close}}
-            >
-                <Button
-                    @layout='fake-link'
-                    {{on 'click' (fn @manager.changeSort 'name')}}
-                >
-                    {{t 'registries.overview.files.sort_by.name'}}
-                </Button>
-                <Button
-                    @layout='fake-link'
-                    {{on 'click' (fn  @manager.changeSort '-name')}}
-                >
-                    {{t 'registries.overview.files.sort_by.-name'}}
-                </Button>
+        {{#if (gt @manager.selectedFiles.length 0)}}
+            <span data-test-file-selected-count>
+                {{t 'osf-components.file-browser.number_selected' numSelected=@manager.selectedFiles.length}}
                 <Button
-                    @layout='fake-link'
-                    {{on 'click' (fn  @manager.changeSort 'modified')}}
+                    data-test-clear-file-selection
+                    data-analytics-name='Clear file selection'
+                    aria-label={{t 'osf-components.file-browser.clear_selection'}}
+                    {{on 'click' @manager.deselectFiles}}
                 >
-                    {{t 'registries.overview.files.sort_by.modified'}}
+                    <FaIcon @icon='times' />
                 </Button>
-                <Button
-                    @layout='fake-link'
-                    {{on 'click' (fn  @manager.changeSort '-modified')}}
+            </span>
+        {{else}}
+            <label>
+                {{t 'general.filter'}}:
+                <Input
+                    data-test-file-search
+                    @type='text'
+                    class='form-control'
+                    placeholder='{{t 'registries.overview.files.filter_placeholder'}}'
+                    @value={{readonly @manager.filter}}
+                    {{on 'input' this.handleInput}}
+                />
+            </label>
+
+            <ResponsiveDropdown
+                local-class='SortDropdown {{if this.isMobile 'Mobile'}}'
+                @renderInPlace={{true}}
+                @buttonStyling={{true}}
+                @horizontalPosition='left'
```

Comment:
This dropdown could also use an aria label.

Reply:
As above, choice for adding the label to the trigger and content is up to you but could be helpful.